### ISSUE TITLE
[BottomSheet] It is not possible to dismiss a bottom sheet using the Z-swipe gesture in the catalog

### ISSUE THREAD

**Issue by jverkoey** [material-components/material-components-ios#4235]
## Reproduction steps

1. Enable VoiceOver.
2. Go to the `Collection View (Short)` demo in MDCDragons.
3. Press the button to present the bottom sheet.
4. Attempt to dismiss the bottom sheet using the two-finger z-swipe back gesture.

## Expected behavior

The bottom sheet is dismissed.

## Actual behavior

The bottom sheet does not dismiss.

## Platform (please complete the following information)

 - Device: iPhone SE
 - OS: 11.3

## Additional context

If you tap on the presented view controller's content and then perform the z-swipe, the bottom sheet will dismiss as expected, but this is an unexpected behavior.

Other examples have accessibility elements in the presented view controller which receive focus on presentation. In these examples, the z-swipe gesture works as expected.

**Comment by andrewoverton** [comment]
@jverkoey I'm not sure what can be done with this one. I put a bunch of dummy implementations of `-accessibilityPerformEscape` in all the view controllers in the hierarchy and the only one that I could get to be called was the one in `MDCBottomSheetController`, which is already implemented and doing what it's supposed to.

I don't really understand why the current behavior is unexpected

**Comment by romoore** [comment]
@andrewoverton Does the Z-gesture work correctly if the user does not tap on any of the content first?

**Comment by andrewoverton** [comment]
The user has to tap on the content view controller to bring it into focus in order for the Z-gesture work

**Comment by romoore** [comment]
Ok, then I believe that's the bug - can we post a `UIAccessibilityScreenChangedNotification ` event and have it focus on the presented sheet?  Would that allow the user to dismiss using the escape gesture without having to tap on the screen first?

**Comment by andrewoverton** [comment]
@jverkoey, what do you think of this?

[These changes](https://github.com/material-components/material-components-ios/compare/develop...andrewoverton:issue-4235?expand=1) lead to desired behavior if the MDCBottomSheetController's contentViewController has a subview that's an accessibilityElement.

I tested it by replacing the contentViewController in [BottomSheetShortCollectionExample](https://github.com/material-components/material-components-ios/blob/develop/components/BottomSheet/examples/BottomSheetShortCollectionExample.m) (a BottomSheetDummyCollectionViewController) with the following: 
```
  UIViewController *viewController = [[UIViewController alloc] init];
  UIButton *button = [[UIButton alloc] initWithFrame:CGRectMake(0, 0, 200, 200)];
  [button setTitle:@"button" forState:UIControlStateNormal];
  [viewController.view addSubview:button];
```
With the above view controller it focused on the button on presentation and the Z gesture worked.

With these changes, if the client would want the Z gesture to work with a BottomSheetDummyCollectionViewController, rather than this silly view controller, they'd have to explicitly make the BottomSheetDummyCollectionViewController's view an accessibility element. Is this okay?

@romoore had talked about introducing an API that allows the client to set an initial accessibility element on presentation. This is another option.

**Comment by romoore** [comment]
Unfortunately, those changes would only work if the contentViewController's view *is* an accessibility element. If it's not (because it's a container), then the changes would not work as expected.

To expand on my suggestion, I think we should consider the following:

1. MDCBottomSheetController by default will change the focus to the Scrim (making it an accessibility element with a "Close" label).
2. I see two options here:
    1. Provide an API on MDCBottomSheetController to allow clients to set the initial accessibility focus when the sheet is presented. 
    1. Document that clients need to post a UIAccessibilityScreenChanged notification and reference their desired focus target after the bottom sheet has been presented. I'm not sure if there is a callback API that we could use for this.

**Comment by andrewoverton** [comment]
@romoore - Did you test it with the 4 line view controller I wrote out in my last comment? Based off my testing, the contentViewController does not need to be an accessibility element. It can also just have a subview that's an accessibility element. In the case of the simple UIViewController with a UIButton subview that I outlined above, the contentViewController is not an accessibility element, but its UIButton subview is. When the bottom sheet is presented, the "screen changed" notification I post (with the contentViewController's view--not its UIButton subview--as an argument) results in the button becoming focused. The Z gesture works as planned. Near the end of my comment I mentioned that for the current example to work we'd need to explicitly set the BottomSheetDummyCollectionViewController's view to be an accessibility element. I _think_ this is because it A) isn't an accessibility element, and B) doesn't have any subviews that are accessibility elements.. 

I like your suggestions. Should we do all of them or just some of them or...?

**Comment by jverkoey** [comment]
Is the essential discovery here that so long as there is at least one accessibility element in the bottom sheet, it can be dismissed with the z-swipe?

**Comment by andrewoverton** [comment]
@jverkoey, yes, exactly. And my understanding earlier was slightly off--it has nothing to do with that diff I linked to haha. If there's anything accessible in the presented view it gets focus. That simple. With that in mind, what should we do? Simply make the collection view dummy view controller's view an accessibility element and document that this is how presenting view controllers works with VoiceOver on? Is that even our responsibility to document?

**Comment by jverkoey** [comment]
So the inversion of that understanding is that this bug **only** occurs if the client presents content that is completely non-interactive. Is that correct?

**Comment by andrewoverton** [comment]
I think so, yep

**Comment by andrewoverton** [comment]
Can we close this one? As long as there is something accessible in the presented view the presented view gets focus and the Z gesture works. Is that good enough for us?

**Comment by romoore** [comment]
Sure, let's close it and we can reopen if someone feels additional guidance to clients is needed.

**Comment by andrewoverton** [comment]
Thanks @romoore
